File: 3471-minimum-average-of-smallest-and-largest-elements/3471-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        double min_avg = Double.POSITIVE_INFINITY;
        Arrays.sort(nums);
        // double avg = 0;
        // int len = nums.length/2;
        for(int i = 0;i<nums.length/2;i++){
            double avg = (nums[i] + nums[nums.length-1-i]) / 2.0;
            min_avg = Math.min(min_avg, avg);
        }
        return min_avg;
    }
}

File: 3471-minimum-average-of-smallest-and-largest-elements/3471-minimum-average-of-smallest-and-largest-elements.py
class Solution(object):
    def minimumAverage(self, nums):
        """
        :type nums: List[int]
        :rtype: float
        """
        min_avg = float('inf')
        nums.sort()
        for i in range(len(nums)/2):
            min_avg = min(min_avg, (nums[i] + nums[-1-i])/2.0)
        return min_avg
        